runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngineConfig.kt

```diff
@@ -8,6 +8,11 @@ package aws.smithy.kotlin.runtime.http.engine.okhttp
 import aws.smithy.kotlin.runtime.http.engine.HttpClientEngineConfig
 
 public class OkHttpEngineConfig private constructor(builder: Builder) : HttpClientEngineConfig(builder) {
+    /**
+     * The maximum number of connections to open to a single host.
+     */
+    public val maxConnectionsPerHost: UInt = builder.maxConnectionsPerHost
```

Comment:
Question: Should this be a knob on base config? I believe it's called out in SRA as `maxConcurrency` (though this name is more appropriate I think)

Reply:
I didn't see an equivalent setting for CRT engine so I added it to OkHttp specifically. If you know of a CRT setting that maps (or if we're okay with this being ignored by CRT) then I have no problem relocating it.

Reply:
I think we can reasonably support this in CRT.

Connection managers are [already per/host in CRT engine](https://github.com/awslabs/smithy-kotlin/blob/66e95fb45a9a508677eac67dcbd210820eb4dea7/runtime/protocol/http-client-engines/http-client-engine-crt/common/src/aws/smithy/kotlin/runtime/http/engine/crt/CrtHttpEngine.kt#L83). Would we just want to take either the max/min of `maxConnections` and `maxConnectionsPerHost` for CRT?